File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int [] output = new int[nums.length];
        for (int i = 0; i < nums.length; i++){
            int left [] = Arrays.copyOfRange(nums, 0, i);
            int right [] = Arrays.copyOfRange(nums, i + 1, nums.length);
            int leftProduct = 1;
            int rightProduct = 1;
            for (int l = 0; l < left.length; l++){
                leftProduct = left[l] * leftProduct;
            }
            for (int r = 0; r < right.length; r++){
                rightProduct = right[r] * rightProduct;
            }
            output[i] = leftProduct * rightProduct;
        }
        return output;

    }
}  

File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        res = 0
        #left and right pointers to show what part of array is in current "level"
        l, r = 0, 0

        while r < len(nums) - 1:
            #how far can the next level go 
            farthest = 0
            #set farthest variable equal to largest combo of jump distance and current index
            for i in range(l, r + 1):
                farthest = max(farthest, i + nums[i])
            #move pointers
            l = r + 1
            r = farthest
            #add one to res every iteration so you know how many "levels" there are
            res += 1
        return res


        